### DISCUSSION TITLE
which parameter

### DISCUSSION THREAD

**Discussion by D19CrMei** [talmolab/sleap#878] [Help!]
In my experiment, because the camera is too close to the target, the detected object is very large. When the skeleton is calibrated and trained, the skeleton in the prediction result is incomplete, and some joints do not appear. Which parameters can be modified to output the target completely? thanks
![image](https://user-images.githubusercontent.com/73282211/182029921-29b6897b-aa95-4468-8442-2dc5677207da.png)
![image](https://user-images.githubusercontent.com/73282211/182029926-5fa4a027-cab0-43d2-bbd6-bb63713714aa.png)

**Comment by roomrys** [comment]
Hi @D19CrMei,

It sounds like you need to increase your receptive field size. There are some good pointers in this [discussion](https://github.com/talmolab/sleap/discussions/854#discussioncomment-3255801), namely the second item
> 2. The animal is large relative to the field of view.
In this case, it might be that the model has a hard time learning to reason about information in parts of the image that are far apart. This can be addressed by:
Decreasing the input scale to 0.5. This will make a lot of things faster and easier, but it might not be possible if the smallest features on your images (probably the leg tips) are already very small (2-4 px wide).
Increasing the receptive field size of the model. This can be done by increasing the max stride of the network to 32 or 64. This will increase the model size significantly, which means you might run out of memory, so you can balance that out by decreasing the filter rate to 1.5.

Let us know if increasing the receptive field size (either by increasing the **max stride** or decreasing the **input scaling**) helps.

Thanks,
Liezl

**Reply by D19CrMei** [reply]
1. I tried the parameters and still there are several nodes that cannot be detected. My parameter settings are as follows:
Input scaling = 1.0,  crop size =480,  max stride =32,  filters=24,  filters rate = 1.5,  sigma = 2.5,  output stride =4 .
But when I set max stride =32, the operation will report an error.
2. There are three pages during training in the GUI. Is there any difference in the parameter settings of the last two pages?
3. I am analyzing now, I need to increase the receptive field, because my data occupies most of the lower left corner. Can you elaborate more about the parameter part? thanks

**Reply by roomrys** [reply]
Hi @D19CrMei,

1. Can you also provide the error log?
2. Each page after the first page in the Training Pipeline sets the hyperparameters for that specific model (one model per page). The preprocessing should always be set the same, but the other parameters can be changed per model to increase performance. Set the RF to be about the size of the animal for the centroid model. For the centered instance model, the RF usually does not need to be changed.
3. Do you mind sharing a screenshot of the entire training pipeline (including the leftmost section that shows the receptive field (RF) overlaid on a sample image)?  A visual of your data vs the RF would be very helpful if that is possible. To increase the RF, you can either increase the max stride of decrease the input scaling. 

Thanks,
Liezl
